cdap-ui/app/cdap/components/FieldLevelLineage/v2/LineageSummary/index.tsx

```diff
@@ -28,40 +30,151 @@ const styles = (theme) => {
       display: 'flex',
       justifyContent: 'space-between',
     },
+    container: {
+      position: 'absolute' as 'absolute',
+      height: '100%',
+      width: '100%',
+    },
   };
 };
 
-function LineageSummary({ classes }) {
-  return (
-    <Consumer>
-      {({ causeSets, target, targetFields, impactSets, firstCause, firstImpact, firstField }) => {
-        return (
-          <div className={classes.root}>
-            <div>
-              <FllHeader type="cause" first={firstCause} total={Object.keys(causeSets).length} />
-              {Object.keys(causeSets).map((key) => {
-                return <FllTable key={key} tableName={key} fields={causeSets[key]} />;
-              })}
-            </div>
-            <div>
-              <FllHeader
-                type="target"
-                first={firstField}
-                total={Object.keys(targetFields).length}
-              />
-              <FllTable key={target} tableName={target} fields={targetFields} />
-            </div>
-            <div>
-              <FllHeader type="impact" first={firstImpact} total={Object.keys(impactSets).length} />
-              {Object.keys(impactSets).map((key) => {
-                return <FllTable key={key} tableName={key} fields={impactSets[key]} />;
-              })}
+class LineageSummary extends React.Component<{ classes }> {
+  private activeLinks;
+
+  private drawLineFromLink({ source, destination }) {
+    // get source and destination elements and their coordinates
+    const sourceEl = d3.select(`#${source}`);
+    const destEl = d3.select(`#${destination}`);
+
+    const offsetX = -100; // From the padding on the LineageSummary
+    const offsetY = -50; // From the FllHeader
+
+    const sourceXY = sourceEl.node().getBoundingClientRect();
+    const destXY = destEl.node().getBoundingClientRect();
+
+    const sourceX1 = sourceXY.right + offsetX;
+    const sourceY1 = sourceXY.top + offsetY + 0.5 * sourceXY.height;
+    const sourceX2 = destXY.left + offsetX;
+    const sourceY2 = destXY.top + offsetY + 0.5 * sourceXY.height;
+
+    // draw an edge from line start to line end
+    const linkContainer = d3.select(`#${source}_${destination}`);
+
+    const third = (sourceX2 - sourceX1) / 3;
+
+    // Draw a line with a bit of curve between the straight parts
+    const lineGenerator = d3.line().curve(d3.curveMonotoneX);
+
+    const points = [
+      [sourceX1, sourceY1],
+      [sourceX1 + third, sourceY1],
+      [sourceX2 - third, sourceY2],
+      [sourceX2, sourceY2],
+    ];
+
+    linkContainer
+      .append('path')
+      .style('stroke', '#bbbbbb')
```

Comment:
nit: Can this come from theme?

Reply:
I'll be addressing this in the next PR, and I've added a to do for now.

Reply:
Just updated the colors to come from ThemeWrapper/colors after merging the PR for merging colors from ThemeWrapper into separate file.